src/npyffi/array.rs

```diff
@@ -31,28 +31,35 @@ pub static PY_ARRAY_API: PyArrayAPI = PyArrayAPI::new();
 
 /// See [PY_ARRAY_API] for more.
 pub struct PyArrayAPI {
-    api: Cell<*const *const c_void>,
+    api: AtomicUsize,
 }
 
 impl PyArrayAPI {
     const fn new() -> Self {
         Self {
-            api: Cell::new(ptr::null_mut()),
+            api: AtomicUsize::new(0),
```

Comment:
`Usize` may not be the same as pointer size on all platforms, should this use `AtomicPtr`?

Reply:
It can certainly use `AtomicPtr` (and should probably do so just for readability). Can you elaborate on platforms where `usize` maybe larger than a pointer? Are you thinking of CHERI and friends?

Reply:
Fix pushed.

Reply:
👍 essentially just feeling inspired to have good hygiene about this after this recent stack IRLO thread: https://internals.rust-lang.org/t/pre-rfc-usize-is-not-size-t/15369